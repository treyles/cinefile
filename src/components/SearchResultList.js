import React from 'react';
import PropTypes from 'prop-types';

export default class SearchResultList extends React.Component {
  renderMatchLi(media) {
    const { handleSelect } = this.props;

    const inLibraryColor = media.inLibrary ? 'inLibraryColor' : '';

    const inLibraryTag = (
      <div className="inlibrary-tag">
        <h2>{media.inLibrary && 'Delete from library'}</h2>
      </div>
    );

    const mediatypeTag = (
      <div className="mediatype-tag">
        <h2>{media.media_type === 'tv' ? 'Television' : 'Movie'}</h2>
      </div>
    );

    return (
      <li
        key={media.id}
        className={media.selected ? 'active' : ''}
        onClick={() => handleSelect(media, media.inLibrary)}
      >
        <img
          className="search-poster"
          src={`https://image.tmdb.org/t/p/w92${media.poster_path}`}
          alt={`${media.title ? media.title : media.name} poster`}
        />
        <div className="search-info">
          <h2 className={inLibraryColor}>
            {media.title ? media.title : media.name}
          </h2>
          <h3 className={inLibraryColor}>
            {typeof media.release_date === 'string'
              ? media.release_date.substring(0, 4)
              : media.first_air_date.substring(0, 4)}
          </h3>
        </div>
        {media.inLibrary ? inLibraryTag : mediatypeTag}
      </li>
    );
  }

  render() {
    const { matches } = this.props;

    return (
      <div className="search-results">
        <ul>
          {matches.length ? (
            matches.map(media => this.renderMatchLi(media))
          ) : (
            <li className="zero-results">
              <h2>Your search returned zero results</h2>
            </li>
          )}
        </ul>
      </div>
    );
  }
}

SearchResultList.propTypes = {
  matches: PropTypes.arrayOf(PropTypes.object).isRequired
};
