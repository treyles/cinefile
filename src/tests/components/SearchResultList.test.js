import React from 'react';
import { shallow } from 'enzyme';
import SearchResultList from '../../components/SearchResultList';

const movieResult = [{
  id: 'abc',
  title: 'Contact',
  release_date: '1997-10-30',
  media_type: 'movie'
}]

const tvResult = [{
  id: '123',
  name: 'Stranger Things',
  release_date: '2016-10-30',
  media_type: 'tv'
}]

describe('SearchResultList', () => {
  let component;

  beforeEach(() => {
    component = shallow( <SearchResultList matches={movieResult} /> )
  })

  it('should render correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('should render correct length of matches', () => {
    component.setProps({
      matches: movieResult.concat(tvResult)
    })

    expect(component.find('li')).toHaveLength(2)
  })

  it('should render movie title if movie', () => {
    component.setProps({
      matches: movieResult
    })

    expect(component.find('li h2').first().text()).toEqual('Contact')
  })

  it('should render tv title if tv', () => {
    component.setProps({
      matches: tvResult
    })

    expect(component.find('li h2').first().text()).toEqual('Stranger Things')
  })

  it('should render year', () => {
    expect(component.find('li h3').first().text()).toEqual('1997')
  })

  it('should render media tag as `Television`', () => {
    component.setProps({
      matches: tvResult
    })

    expect(component.find('.mediatype-tag')).toHaveLength(1)
    expect(component.find('.mediatype-tag').text()).toEqual('Television')
  })

  it('should render media tag as `Movie`', () => {
    component.setProps({
      matches: movieResult
    })

    expect(component.find('.mediatype-tag')).toHaveLength(1)
    expect(component.find('.mediatype-tag').text()).toEqual('Movie')
  })

  it('should render `Delete from library` button if already in library', () => {
    component.setProps({
      matches: [ Object.assign(tvResult[0], {
        inLibrary: true
      })]
    })

    expect(component.find('.mediatype-tag')).toHaveLength(0)
    expect(component.find('.inlibrary-tag')).toHaveLength(1)
  })
})