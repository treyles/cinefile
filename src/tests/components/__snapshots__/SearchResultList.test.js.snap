// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultList
    matches={
      Array [
        Object {
          "id": "abc",
          "media_type": "movie",
          "release_date": "1997-10-30",
          "title": "Contact",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li
          className=""
          onClick={[Function]}
        >
          <img
            alt="Contact poster"
            className="search-poster"
            src="https://image.tmdb.org/t/p/w92undefined"
          />
          <div
            className="search-info"
          >
            <h2
              className=""
            >
              Contact
            </h2>
            <h3
              className=""
            >
              1997
            </h3>
          </div>
          <div
            className="mediatype-tag"
          >
            <h2>
              Movie
            </h2>
          </div>
        </li>
      </ul>,
      "className": "search-results",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className=""
            onClick={[Function]}
          >
            <img
              alt="Contact poster"
              className="search-poster"
              src="https://image.tmdb.org/t/p/w92undefined"
            />
            <div
              className="search-info"
            >
              <h2
                className=""
              >
                Contact
              </h2>
              <h3
                className=""
              >
                1997
              </h3>
            </div>
            <div
              className="mediatype-tag"
            >
              <h2>
                Movie
              </h2>
            </div>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "abc",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Contact poster"
                className="search-poster"
                src="https://image.tmdb.org/t/p/w92undefined"
              />,
              <div
                className="search-info"
              >
                <h2
                  className=""
                >
                  Contact
                </h2>
                <h3
                  className=""
                >
                  1997
                </h3>
              </div>,
              <div
                className="mediatype-tag"
              >
                <h2>
                  Movie
                </h2>
              </div>,
            ],
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Contact poster",
                "className": "search-poster",
                "src": "https://image.tmdb.org/t/p/w92undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className=""
                  >
                    Contact
                  </h2>,
                  <h3
                    className=""
                  >
                    1997
                  </h3>,
                ],
                "className": "search-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Contact",
                    "className": "",
                  },
                  "ref": null,
                  "rendered": "Contact",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1997",
                    "className": "",
                  },
                  "ref": null,
                  "rendered": "1997",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                  Movie
                </h2>,
                "className": "mediatype-tag",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Movie",
                },
                "ref": null,
                "rendered": "Movie",
                "type": "h2",
              },
              "type": "div",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li
            className=""
            onClick={[Function]}
          >
            <img
              alt="Contact poster"
              className="search-poster"
              src="https://image.tmdb.org/t/p/w92undefined"
            />
            <div
              className="search-info"
            >
              <h2
                className=""
              >
                Contact
              </h2>
              <h3
                className=""
              >
                1997
              </h3>
            </div>
            <div
              className="mediatype-tag"
            >
              <h2>
                Movie
              </h2>
            </div>
          </li>
        </ul>,
        "className": "search-results",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className=""
              onClick={[Function]}
            >
              <img
                alt="Contact poster"
                className="search-poster"
                src="https://image.tmdb.org/t/p/w92undefined"
              />
              <div
                className="search-info"
              >
                <h2
                  className=""
                >
                  Contact
                </h2>
                <h3
                  className=""
                >
                  1997
                </h3>
              </div>
              <div
                className="mediatype-tag"
              >
                <h2>
                  Movie
                </h2>
              </div>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "abc",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Contact poster"
                  className="search-poster"
                  src="https://image.tmdb.org/t/p/w92undefined"
                />,
                <div
                  className="search-info"
                >
                  <h2
                    className=""
                  >
                    Contact
                  </h2>
                  <h3
                    className=""
                  >
                    1997
                  </h3>
                </div>,
                <div
                  className="mediatype-tag"
                >
                  <h2>
                    Movie
                  </h2>
                </div>,
              ],
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Contact poster",
                  "className": "search-poster",
                  "src": "https://image.tmdb.org/t/p/w92undefined",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2
                      className=""
                    >
                      Contact
                    </h2>,
                    <h3
                      className=""
                    >
                      1997
                    </h3>,
                  ],
                  "className": "search-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Contact",
                      "className": "",
                    },
                    "ref": null,
                    "rendered": "Contact",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1997",
                      "className": "",
                    },
                    "ref": null,
                    "rendered": "1997",
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h2>
                    Movie
                  </h2>,
                  "className": "mediatype-tag",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Movie",
                  },
                  "ref": null,
                  "rendered": "Movie",
                  "type": "h2",
                },
                "type": "div",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
